editor/sidebar/post-visibility/index.js

```diff
@@ -111,12 +129,17 @@ class PostVisibility extends Component {
 					aria-expanded={ this.state.opened }
 					className="editor-post-visibility__toggle button-link"
 					onClick={ this.toggleDialog }
+					ref={ toggle => this.toggle = toggle }
 				>
 					{ getVisibilityLabel( visibility ) }
 				</button>
 
 				{ this.state.opened &&
-					<div className="editor-post-visibility__dialog">
+					<div className="editor-post-visibility__dialog"
+						onKeyDown={ this.handleKeyDown }
```

Comment:
I think this rule should be improved a bit. While the intent is good, it should be allowed to use keyboard events this way when one knows what he's doing. Also, why doesn't catch `onBlur` then? Also onBlur can indicate keyboard interaction. 
https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-static-element-interactions.md

Also, some of the "solutions" mentioned in the docs, like adding `role="presentation"` to elements that already have no semantic meaning, don't look right to me and are against the ARIA spec.

Will disable the rule (also remove the previous disabling and comment for htmlFor).